libchisel/src/gc.rs

```diff
@@ -0,0 +1,150 @@
+extern crate wasm_gc;
+extern crate parity_wasm;
+
+use super::ModuleTranslator;
+use parity_wasm::elements::*;
+
+pub struct WasmGC;
+
+impl WasmGC {
+    pub fn new() -> Self {
+        WasmGC{}
+    }
+}
+
+impl ModuleTranslator for WasmGC {
+    fn translate(self, module: &mut Module) -> Result<bool, String> {
+        let serialized = parity_wasm::elements::serialize::<Module>(module.clone()).expect("Could not serialize module");
+        match wasm_gc::Config::new().gc(&serialized[..]) {
+            Ok(gced_bytes) => {
+                let gced = parity_wasm::elements::deserialize_buffer::<Module>(&gced_bytes[..]).expect("Could not deserialize gc'ed module");
```

Comment:
Why not just return the modified version?

Reply:
because the `Translator` interface doesn't return a module, and I am passed a mutable reference. We could change the interface to `Result<Module>` if it makes sense.

Reply:
Perhaps due to limitations in pwasm it would make sense to have two `translate` functions, one which returns a new instance and the current one which takes a mutable module.

The CLI then could try calling both and see which one succeeded. Or we could add helpers which rewrites a `&mut Module` with a given `Module`. (This is what @gballet is doing internally.)

@jakelang what do you think?

Reply:
I agree with this, or just choosing the new module route because handling lifetimes through this API makes for some inflexibility.

Reply:
I still don't get why aren't we using this proposed API as opposed to this manual labour of changing mutable data?

Reply:
decided offline: making this change after this is merged

Reply:
yeah I think that returning a module is the best thing, because otherwise that's a lot of data copied in vain.